hw/ip/pattgen/data/pattgen.hjson

```diff
@@ -0,0 +1,160 @@
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          resval: "0",
+          name: "ENABLE_CH0",
+          desc: "Enable pattern generator functionality for channel 0"
+        }
+        { bits: "1",
+          resval: "0",
+          name: "ENABLE_CH1",
+          desc: "Enable pattern generator functionality for channel 1"
+        }
+        { bits: "2",
+          resval: "0",
+          name: "POLARITY_CH0",
+          desc: "Clock (`pcl`) polarity for channel 0.  If high, `pda` signal changes on falling edge of pcl signal, otherwise pda changes on rising edge"
+        }
+        { bits: "3",
+          resval: "0",
+          name: "POLARITY_CH1",
+	  desc: "Clock (`pcl`) polarity for channel 1.  If high, `pda` signal changes on falling edge of `pcl` signal, otherwise pda changes on rising edge"
+
+          desc: "Enable pattern generator functionality for channel 0"
+        }
+      ]
+    }
+    { name: "PATT_PREDIV_CH0",
+      desc: "PATTGEN pre-divider multi-registers for channel 0",
+      count: "NumRegsPrediv",
+      cname: "PATTGEN",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "31:0",
+          resval: "0",
+          name: "CLK_RATIO",
+          desc: "Divider ratio from based I/O clock for channel 0"
+        }
+      ]
+    },
+    { name: "PATT_PREDIV_CH1",
+      desc: "PATTGEN pre-divider multi-registers for channel 1",
+      count: "NumRegsPrediv",
+      cname: "PATTGEN",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "31:0",
+          resval: "0",
+          name: "CLK_RATIO",
+          desc: "Divider ratio from based I/O clock for channel 1"
+        }
+      ]
+    },
+    { multireg: {
+        name: "PATT_DATA_CH0",
+        desc: "PATTGEN Seed pattern multi-registers for channel 0",
+        count: "NumRegsData",
+        cname: "PATTGEN",
+        swaccess: "wo",
+        hwaccess: "hro",
+        hwqe: "true",
+        fields: [
+          { bits: "31:0",
+            resval: "0",
+            name: "DATA",
+            desc: "Seed pattern for channel 0 (2-64 bits)"
+          }
+        ]
+      }
+    },
+    { multireg: {
+        name: "PATT_DATA_CH1",
+        desc: "PATTGEN Seed pattern multi-registers for channel 1",
+        count: "NumRegsData",
+        cname: "PATTGEN",
+        swaccess: "wo",
+        hwaccess: "hro",
+        hwqe: "true",
+        fields: [
+          { bits: "31:0",
+            resval: "0",
+            name: "DATA",
+            desc: "Seed pattern for channel 1 (2-64 bits)"
+          }
+        ]
+      }
+    },
+    { name: "PATT_SIZE",
+      desc: "PATTGEN pattern length",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "5:0",
+          resval: "0",
+          name: "LEN_CH0",
+          desc: "Length of the seed pattern for channel 0 (1-64 bits), minus 1"
```

Comment:
```suggestion
          desc: "Length of the seed pattern for Channel 0, minus 1. Valid values: 0..63."
```

* For consistency: "Channel N" is capitalized in the doc, either capitalize it here as well, or change the doc.
* I find the use of `-` often confusing in docs: is it a minus, or a range, or something else? 


(This comment also applies to the other registers below/above.)

Reply:
The dash is a range.  Your point about it being possibly ambiguous is noted, and I can make this change. 

  However, there is an open question that I'd like to pick your brain about.   In non-technical English the dash is the standard way of expressing ranges, but you are suggesting a perl-ish/rust-ish range indication.  This brings up a question of what the intended coding specs should be for register usage spec.   For instance in another spec @cdgori suggested we use verilog syntax for hex numbers, as opposed to C syntax (which would be appropriate for those writing DIFs).    Is there a preference as to what technical language to use?

Reply:
FYI, I think (if I remember the situation) I said that because I was trying to get the exact width of quantity expressed in the spec, which can be done with Verilog syntax but not C syntax (for non-nibble-aligned sizes).